TempleEight/docs/guide/orchestration.md

```diff
@@ -3,3 +3,172 @@ id: orchestration
 title: Orchestration
 sidebar_label: Orchestration
 ---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+
+With distributed systems, equally important as the application is the way that it's orchestrated. 
+Temple provides several industry standards methods of automatically deploying your services.
+
+## Templefile Changes
+
+Orchestrating your project automatically requires only a single line of code in your Templefile.
+Working from the example in the [Getting Started](../getting-started) guide:
+
+```templefile {4}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+We can see here that the Templefile features a `#provider` annotation in the project block.
+This is how we specify that our project uses orchestration, the name in the argument specifies which framework we are generating for.
+
+The currently supported frameworks are:
+
+* Docker Compose
+
+* Kubernetes
+
+All of the frameworks are generated to share a common toolchain, so that usage is the same no matter which you choose.
+
+## Kong API Gateway
+
+An API Gateway is an infrastructure component designed to be the entry point to your application. 
+It receives all requests from the user, performs actions like verifying their authentication, and then forwards the request to the correct microservice.
+
+Temple makes use of the existing [Kong API Gateway](https://docs.konghq.com/2.0.x/getting-started/quickstart/) for this purpose, and automatically generates all the configuration it requires.
+
+The only considerations you as the user needs to make is that when you try to make requests to any of your deployed services, the requests need to be addressed to Kong's URL.
```

Comment:
```suggestion
An API Gateway provides a single entry point into your project infrastructure, meaning it can direct requests to deployed services whether they are on a single machine or multiple machines.
To do this, requests need to be addressed to Kong's URL.
```